src/tool.cpp

```diff
@@ -416,8 +416,7 @@ DigParams getDigParams(const ItemGroupList &groups,
 			// The actual number of uses increases
 			// exponentially with leveldiff.
 			// If the levels are equal, real_uses equals cap.uses.
-			u32 real_uses = cap.uses * pow(3.0, leveldiff);
-			real_uses = MYMIN(real_uses, U16_MAX);
+			const u32 real_uses = MYMIN(cap.uses * pow(3.0, leveldiff), U16_MAX);
```

Comment:
> "Fix unsigned number wraparound" (https://github.com/minetest/minetest/pull/14365/commits/85e9d137d94ad9e0749ca069a6a42d0615deb71f)

Nitpick: Unsigned number wraparound is not undefined behavior. The problem here is that due to the `pow(3.0, blah)`, what we really have is a double, and that might not fit in `u32`, which would be UB. So we need to limit the double before we try to convert it to `u32`. The fix should work, though I feel an explicit `std::min<f64>` or similar might be more readable.

Reply:
Yeah, that's right. I do not have a strong opinion if I should keep these changes or remove them.

Reply:
No, this change is good. It makes the code slightly more concise and fixes undefined behavior. I'm just saying the reason for changing this is a different one, and that perhaps it would be helpful to use `std::min` to let you be explicit about the types.

Reply:
Now I see, I agree that `std::min<f64>` makes sense, so I applied it.